sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_http_logging_policy.py

```diff
@@ -41,213 +43,303 @@
+    return ret_str
 
 
 class CosmosHttpLoggingPolicy(HttpLoggingPolicy):
 
     def __init__(
-        self,
-        logger: Optional[logging.Logger] = None,
-        global_endpoint_manager: Optional[_GlobalEndpointManager] = None,
-        database_account: Optional[DatabaseAccount] = None,
-        *,
-        enable_diagnostics_logging: bool = False,
-        diagnostics_handler: Optional[Union[Callable, Mapping]] = None,
-        **kwargs
+            self,
+            logger: Optional[logging.Logger] = None,
+            global_endpoint_manager: Optional[_GlobalEndpointManager] = None,
+            database_account: Optional[DatabaseAccount] = None,
+            *,
+            enable_diagnostics_logging: bool = False,
+            **kwargs
     ):
         self._enable_diagnostics_logging = enable_diagnostics_logging
-        self.diagnostics_handler = diagnostics_handler
-        self.__request_already_logged = False
-        if self.diagnostics_handler and callable(self.diagnostics_handler):
-            if hasattr(self.diagnostics_handler, '__get__'):
-                self._should_log = types.MethodType(diagnostics_handler, self)  # type: ignore
-            else:
-                self._should_log = self.diagnostics_handler  # type: ignore
-        elif isinstance(self.diagnostics_handler, Mapping):
-            self._should_log = self._dict_should_log  # type: ignore
-        else:
-            self._should_log = self._default_should_log  # type: ignore
         self.__global_endpoint_manager = global_endpoint_manager
-        self.__client_settings = self.__get_client_settings()
         self.__database_account_settings: Optional[DatabaseAccount] = (database_account or
                                                                        self.__get_database_account_settings())
-        self._resource_map = {
-            'docs': 'document',
-            'colls': 'container',
-            'dbs': 'database'
-        }
-        super().__init__(logger, **kwargs)
-        if self._enable_diagnostics_logging:
-            cosmos_disallow_list = ["Authorization", "ProxyAuthorization"]
-            cosmos_allow_list = [
-                v for k, v in HttpHeaders.__dict__.items() if not k.startswith("_") and k not in cosmos_disallow_list
-            ]
-            self.allowed_header_names = set(cosmos_allow_list)
-
-    def on_request(self, request: PipelineRequest[HTTPRequestType]) -> None:
-        verb = request.http_request.method
+
+        self.logger: logging.Logger = logger or logging.getLogger("azure.cosmos.cosmos_http_logging_policy")
+        self.allowed_query_params: Set[str] = set()
+        self.allowed_header_names: Set[str] = set(self.__class__.DEFAULT_HEADERS_ALLOWLIST)
+
+        cosmos_disallow_list = ["Authorization", "ProxyAuthorization", "Transfer-Encoding"]
+        cosmos_allow_list = [
+            v for k, v in HttpHeaders.__dict__.items() if not k.startswith("_") and k not in cosmos_disallow_list
```

Comment:
It might be worth adding an inline comment (not in the public docstring) to the `HttpHeaders` class to inform future devs/maintainers that headers the add to that class will be logged unless they also update the disallow list.